core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2272,6 +2273,25 @@ private[spark] object Utils extends Logging {
     }.map(threadInfoToThreadStackTrace)
   }
 
+  /** Return a heap dump */
+  def getHeapDump: InputStream = {
+    try {
+      val clazz = classOf[HotSpotDiagnosticMXBean]
+      val server = ManagementFactory.getPlatformMBeanServer
+      val bean = ManagementFactory.newPlatformMXBeanProxy(server
+        , "com.sun.management:type=HotSpotDiagnostic"
+        , clazz)
+      val tmpDir = createTempDir("heap-dump")
+      tmpDir.deleteOnExit()
+      val absolutePath = s"${tmpDir.getAbsolutePath}/${System.nanoTime()}.hprof"
```

Comment:
`System.nanoTime()` does not give you a unique value. There is a high risk of name clashing here.

Reply:
Fixed